OAK-11267 - Upgrade Azure SDK V8 to V12 for oak-blob-azure

andreeastroe96 opened the pull request.

kwin commented:
Every larger PR needs a JIRA ticket in https://issues.apache.org/jira/projects/OAK!

ierandra commented on oak-blob-cloud-azure/src/main/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/UtilsV12.java:
```diff
@@ -0,0 +1,107 @@
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public final class UtilsV12 {
```
the classes with version v8 should have in the name V8 and the classes with the new version (v 12) should have the name with no version

andreeastroe96 commented:
> Every larger PR needs a JIRA ticket in https://issues.apache.org/jira/projects/OAK!

andreeastroe96 commented:
> Every larger PR needs a JIRA ticket in https://issues.apache.org/jira/projects/OAK!

Hi! I linked this PR with this JIRA issue: https://issues.apache.org/jira/browse/OAK-11267 . Please let me know if something else is required.

kwin commented:
> > Every larger PR needs a JIRA ticket in https://issues.apache.org/jira/projects/OAK!
> 
> Hi! I linked this PR with this JIRA issue: https://issues.apache.org/jira/browse/OAK-11267 . Please let me know if something else is required.

Please read https://jackrabbit.apache.org/oak/docs/participating.html. Currently the commit message does not reference the Oak issue.

anchela commented:
@kwin , maybe it's just style.... but to me your review comments don't come across as welcoming (see https://apache.org/foundation/policies/conduct). @andreeastroe96 is a new contributor to the project and there is IMHO no need to be angry... a friendly reminder will do magic.

kwin commented:
@anchela You are right.
@andreeastroe96 Sorry, if I came across rude. That was not intended. Just be aware that this is the ASF Jackrabbit context so neither AEM nor GRANITE nor any other restricted references should be used here for everyone to be able to follow/review accordingly. Thanks.

ierandra commented on oak-blob-cloud-azure/src/main/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/AzureBlobContainerProvider.java:
```diff
@@ -171,176 +152,99 @@ public String getContainerName() {
-        new ExecutorCloser(executorService).close();
-        log.info("Refresh token executor service shutdown completed");
+    @NotNull
+    private BlobContainerClient getBlobContainerFromServicePrincipals(String accountName, String clientId, String clientSecret, String tenantId, RequestRetryOptions retryOptions) throws DataStoreException {
```
`clientId `, `clientSecret` and `tenantId` are never used in this method

ierandra replied:
here you are parsing the `accountName` however at line [218](https://github.com/apache/jackrabbit-oak/pull/1861/files#diff-5009e27269a530925293e139d27368a1bc75abb12217bbff287386c4586073b2R218) for method `getBlobContainerFromConnectionString()` you are using the `getContainerName()`.
please use the same logic for both methods

ierandra commented on oak-blob-cloud-azure/src/main/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/Utils.java:
```diff
@@ -1,130 +1,92 @@
-        }
-        return client;
-    }
+    public static BlobContainerClient getBlobContainer(String accountName, String accountKey, @NotNull final String connectionString,
```
`accountName` and `accountKey` are never used in this method.

ierandra commented on oak-blob-cloud-azure/src/main/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/AzureBlobContainerProvider.java:
```diff
@@ -171,176 +152,99 @@ public String getContainerName() {
         if (StringUtils.isNotBlank(azureConnectionString)) {
             log.debug("connecting to azure blob storage via azureConnectionString");
-            return Utils.getBlobContainer(azureConnectionString, containerName, blobRequestOptions);
+            return getBlobContainerFromConnectionString();
```
any reason why this was moved from `Utils` ?

ierandra commented on oak-blob-cloud-azure/src/main/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/AzureBlobStoreBackend.java:
```diff
@@ -47,81 +70,46 @@
     private int concurrentRequestCount = DEFAULT_CONCURRENT_REQUEST_COUNT;
-    private RetryPolicy retryPolicy;
+    private RequestRetryOptions retryOptions;
     private Integer requestTimeout;
```
this is not used anymore.

ierandra commented on oak-blob-cloud-azure/src/main/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/AzureBlobStoreBackend.java:
```diff
@@ -130,36 +118,19 @@ public class AzureBlobStoreBackend extends AbstractSharedBackend {
-        }
-        requestOptions.setConcurrentRequestCount(concurrentRequestCount);
-        if (enableSecondaryLocation) {
-            requestOptions.setLocationMode(LocationMode.PRIMARY_THEN_SECONDARY);
```
I see you removed the secondary location concept. How was this implemented in azure sdk v12 ? because I cannot see it anywhere

ierandra commented on oak-blob-cloud-azure/src/main/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/AzureBlobStoreBackend.java:
```diff
@@ -199,19 +169,18 @@ public void init() throws DataStoreException {
+                        Strings.emptyToNull(properties.getProperty(AzureConstants.PRESIGNED_HTTP_DOWNLOAD_URI_VERIFY_EXISTS)), true);
 
-                enableSecondaryLocation = PropertiesUtil.toBoolean(
+                boolean enableSecondaryLocation = PropertiesUtil.toBoolean(
```
where is this used?

ierandra commented on oak-blob-cloud-azure/src/main/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/AzureBlobStoreBackend.java:
```diff
@@ -171,16 +142,15 @@ public void init() throws DataStoreException {
 
             try {
-                Utils.setProxyIfNeeded(properties);
+                isProxyNeeded = Utils.isProxyNeeded(properties);
```
IMO, this check can be done directly in `Utils` class since is used only there.

ierandra commented on oak-blob-cloud-azure/src/main/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/Utils.java:
```diff
@@ -1,130 +1,92 @@
+            return new RequestRetryOptions(RetryPolicyType.FIXED, 1, 1, null, null, null);
         }
-        return new RetryExponentialRetry(RetryPolicy.DEFAULT_CLIENT_BACKOFF, retries);
+        return new RequestRetryOptions(RetryPolicyType.EXPONENTIAL, retries, (int) Duration.ofSeconds(3).toMillis(), null, null, null);
```
`tryTimeoutInSeconds` should not be hardcoded. it should use the `AzureConstants.AZURE_BLOB_REQUEST_TIMEOUT)`

ierandra commented on oak-blob-cloud-azure/src/main/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/Utils.java:
```diff
@@ -1,130 +1,92 @@
+                        .proxy(proxyOptions)
+                        .build();
+
+                BlobServiceClient blobServiceClient = new BlobServiceClientBuilder()
```
lines 49-55 are duplicates with 58-63. the method should be refactored and apply some abstractisation.

ierandra commented on oak-blob-cloud-azure/src/main/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/Utils.java:
```diff
@@ -1,130 +1,92 @@
+                                                       boolean isProxyNeeded, final Properties properties) throws DataStoreException {
+        BlobContainerClient blobContainerClient;
+        try {
+            if(isProxyNeeded) {
```
why calling the `isProxyNeeded` from `AzureBlobStoreBackend` and propagating  `isProxyNeeded` all the way when you can call the method directly here.

ierandra commented on oak-blob-cloud-azure/src/main/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/Utils.java:
```diff
@@ -1,130 +1,92 @@
     }
 
-    public static void setProxyIfNeeded(final Properties properties) {
+    public static boolean isProxyNeeded(final Properties properties) {
```
the method should return `ProxyOptions` if true or `null` or false.

ierandra commented on oak-blob-cloud-azure/src/main/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/AzureBlobStoreBackend.java:
```diff
@@ -199,19 +169,18 @@ public void init() throws DataStoreException {
 
-                retryPolicy = Utils.getRetryPolicy(properties.getProperty(AzureConstants.AZURE_BLOB_MAX_REQUEST_RETRY));
+                retryOptions = Utils.getRetryOptions(properties.getProperty(AzureConstants.AZURE_BLOB_MAX_REQUEST_RETRY));
                 if (properties.getProperty(AzureConstants.AZURE_BLOB_REQUEST_TIMEOUT) != null) {
```
why is this check here since `requestTimeout` is not use anywhere?

ierandra commented on oak-blob-cloud-azure/src/main/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/AzureBlobStoreBackend.java:
```diff
@@ -324,46 +299,28 @@ public void write(DataIdentifier identifier, File file) throws DataStoreExceptio
                 addLastModified(blob);
-
-                BlobRequestOptions options = new BlobRequestOptions();
-                options.setConcurrentRequestCount(concurrentRequestCount);
```
you removed this and didn't replaced in azure sdk 12. you can use `com.azure.storage.blob.models.ParallelTransferOptions` and `BlockBlobClient.uploadWithResponse`

ierandra commented on oak-blob-cloud-azure/src/main/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/AzureBlobStoreBackend.java:
```diff
@@ -807,22 +677,26 @@ private static String stripMetaKeyPrefix(String name) {
 
-    private static void addLastModified(CloudBlockBlob blob) {
-        blob.getMetadata().put(LAST_MODIFIED_KEY, String.valueOf(System.currentTimeMillis()));
+    private static void addLastModified(BlockBlobClient blockBlobClient) {
```
the method name is not what actually does

ierandra commented on oak-blob-cloud-azure/src/main/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/AzureBlobStoreBackend.java:
```diff
@@ -658,29 +545,23 @@ public List<DataRecord> getAllMetadataRecords(String prefix) {
-                            blob.getProperties().getLength(),
-                            true));
-                }
+            for (BlobItem blobItem : getAzureContainer().listBlobsByHierarchy(META_DIR_NAME)) {
```
`listBlobsByHierarchy` is listing only the hierarchy, so `META/`.
this has to be changed with
```
ListBlobsOptions listBlobsOptions = new ListBlobsOptions();
listBlobsOptions.setPrefix(META_DIR_NAME);
getAzureContainer().listBlobs(listBlobsOptions, null);
```

ierandra commented on oak-blob-cloud-azure/src/main/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/AzureBlobStoreBackend.java:
```diff
@@ -727,26 +604,22 @@ public void deleteAllMetadataRecords(String prefix) {
-                        total++;
-                    }
+
+            for (BlobItem blobItem : getAzureContainer().listBlobsByHierarchy(META_DIR_NAME)) {
```
same as above:
`listBlobsByHierarchy` is listing only the hierarchy, so `META/`.
this has to be changed with
```
ListBlobsOptions listBlobsOptions = new ListBlobsOptions();
listBlobsOptions.setPrefix(META_DIR_NAME);
getAzureContainer().listBlobs(listBlobsOptions, null);
```

ierandra commented on oak-blob-cloud-azure/src/main/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/AzureBlobStoreBackend.java:
```diff
@@ -874,24 +748,23 @@ URI createHttpDownloadURI(@NotNull DataIdentifier identifier,
-                        EnumSet.of(SharedAccessBlobPermissions.READ),
+                        new BlobSasPermission().setReadPermission(true),
                         httpDownloadURIExpirySeconds,
-                        headers,
```
why was this removed?

andreeastroe96 replied:
According to the comments on the V8 version:    

 /* set empty headers as blank string due to a bug in Azure SDK
     * Azure SDK considers null headers as 'null' string which corrupts the string to sign and generates an invalid
     * sas token
     * */
     
  those headers were empty due to a bug in the Azure SDK. 
  I read in the documentation that the bug was solved in azure sdk 12, therefore, I thought it would not be a breaking change to remove those headers since there was no actual logic implemented around them

ierandra commented on oak-blob-cloud-azure/src/main/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/AzureBlobStoreBackend.java:
```diff
@@ -874,24 +748,23 @@ URI createHttpDownloadURI(@NotNull DataIdentifier identifier,
                 String key = getKeyName(identifier);
-                SharedAccessBlobHeaders headers = new SharedAccessBlobHeaders();
-                headers.setCacheControl(String.format("private, max-age=%d, immutable", httpDownloadURIExpirySeconds));
+                BlobHttpHeaders headers = new BlobHttpHeaders()
```
where is this used since you removed it at line 894?

ierandra commented on oak-blob-cloud-azure/src/main/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/AzureBlobStoreBackend.java:
```diff
@@ -1260,41 +1117,19 @@ protected T computeNext() {
             long start = System.currentTimeMillis();
-            ClassLoader contextClassLoader = currentThread().getContextClassLoader();
-            try {
-                currentThread().setContextClassLoader(getClass().getClassLoader());
```
why did you removed all this logic?

ierandra commented on oak-blob-cloud-azure/src/test/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/UtilsTest.java:
There should be a `UtilsTest.java` and a `UtilsV8Test.java` for both implementation.

andreeastroe96 committed:
OAK-11267 - Upgrade Azure SDK V8 to V12 for oak-blob-azure

mbaedke committed:
OAK-11199: Java 23: getSubject is supported only if a security manager is allowed (#1891)

Added new class to replace the deprecated API

reschke committed:
Revert "OAK-11269 add dedicated workflow for checking commit messages (#1865)"

This reverts commit 6fa76f87601392adeb69bf033ee80ace9c407e6a.

mbaedke committed:
OAK-11199: Java 23: getSubject is supported only if a security manager is allowed

Fixed file pernissions.

mbaedke committed:
OAK-11199: Java 23: getSubject is supported only if a security manager is allowed

Added missing license header.

mbaedke committed:
OAK-6766: Convert oak-lucene to OSGi R7 annotations (#1352)

Done.

mbaedke committed:
OAK-6772: Convert oak-solr-core to OSGi R7 annotations (#1365)

Done

mbaedke committed:
OAK-6767: Remove felix SCR annotation support from parent pom (#1910)

Done

Commit:
OAK-11317 :removed usage of ImmutableSet from guava

Commit:
OAK-11317 : removed conversion to unmodifiableSet for test classes

Commit:
OAK-11317 : removed conversion to set twice for ExternalGroupPrincipalProviderTest

Commit:
OAK-11317 : removed unused imports

Commit:
OAK-11321 : renoved usage of guava Sets.union

reschke committed:
OAK-11327: Remove usage of Guava ImmutableListMultimap (#1918)

Commit:
OAK-11322 : removed usage of guava Sets.intersection

reschke committed:
OAK-11327: Remove usage of Guava ImmutableListMultimap - fix oak-store-composite ITs

Commit:
OAK-11322 : updated package version to 1.5.0

reschke committed:
OAK-11317: Remove usage of Guava ImmutableSet - restore package version to previous value

Commit:
OAK-11322 : revert package update version

reschke committed:
OAK-11328: Remove usage of Guava ImmutableList (test classes) (#1922)

* OAK-11328: Remove usage of Guava ImmutableList (test classes)

* OAK-11328: Remove usage of Guava ImmutableList (test classes)

Commit:
OAK-11324 : removed usage of Guava's Sets.symmetricDifference

Commit:
OAK-11324 : replaced empty hashset constructor usage wth Set.of

Commit:
OAK-11323 : removed usage of Guava's Sets.difference

Commit:
OAK-11325 : removed usage of Guava's Sets.powerSet

reschke committed:
OAK-11330: Remove usage of Guava ImmutableList (Builder) (#1926)

* OAK-11330: Remove usage of Guava ImmutableList (Builder)

* OAK-11330: Remove usage of Guava ImmutableList (Builder) - fix typo in method name

Commit:
OAK-11331 : removed usage of guava's Sets.filter

Commit:
OAK-11331 : removed usage from LastRecoveryAgent file

Commit:
OAK-11326 : removed the usage of Guava Sets

Commit:
OAK-11334 : removed usage of Guava Lists.reverse

reschke committed:
OAK-11332: Remove usage of Guava ImmutableList (copyOf) (#1929)

reschke committed:
OAK-11337: Update (shaded) Guava to 33.4.0 (#1932)

reschke committed:
OAK-11224: remove guava dependencies from POMs (ongoing) - oak-query-spi

reschke committed:
OAK-11329: oak-auth-ldap: update mina dependency to 2.1.10 (#1933)

reschke committed:
OAK-11311: Remove usage of Guava ImmutableList.of() (#1923)

* OAK-11311: Remove usage of Guava ImmutableList - oak-upgrade

* OAK-11311: Remove usage of Guava ImmutableList - oak-upgrade

* OAK-11311: Remove usage of Guava ImmutableList - oak-store-spi

* OAK-11311: Remove usage of Guava ImmutableList - oak-store-document - tests

* OAK-11311: Remove usage of Guava ImmutableList - oak-store-document

* OAK-11311: Remove usage of Guava ImmutableList - oak-store-document

* OAK-11311: Remove usage of Guava ImmutableList - oak-authorization-cug

* OAK-11311: Remove usage of Guava ImmutableList - oak-store-composite

* OAK-11311: Remove usage of Guava ImmutableList - oak-solr-cote

* OAK-11311: Remove usage of Guava ImmutableList - oak-segment-tar

* OAK-11311: Remove usage of Guava ImmutableList - oak-segment-azure

* OAK-11311: Remove usage of Guava ImmutableList - oak-security-spi - tests

* OAK-11311: Remove usage of Guava ImmutableList - oak-security-spi

* OAK-11311: Remove usage of Guava ImmutableList - remove unintended change in oak-core

* OAK-11311: Remove usage of Guava ImmutableList - oak-search

* OAK-11311: Remove usage of Guava ImmutableList - oak-run-commons

* OAK-11311: Remove usage of Guava ImmutableList - oak-run-commons

* OAK-11311: Remove usage of Guava ImmutableList - oak-run

* OAK-11311: Remove usage of Guava ImmutableList - oak-query-spi

* OAK-11311: Remove usage of Guava ImmutableList - oak-lucene

* OAK-11311: Remove usage of Guava ImmutableList - oak-search - remove duplicate import

* OAK-11311: Remove usage of Guava ImmutableList - oak-store-document - remove unused imports

* OAK-11311: Remove usage of Guava ImmutableList - oak-security-spi - make a few lists unmodifiable

* OAK-11311: Remove usage of Guava ImmutableList - oak-jcr

* OAK-11311: Remove usage of Guava ImmutableList - oak-it

* OAK-11311: Remove usage of Guava ImmutableList - oak-exercise

* OAK-11311: Remove usage of Guava ImmutableList - oak-core-spi

* OAK-11311: Remove usage of Guava ImmutableList - oak-core tests

* OAK-11311: Remove usage of Guava ImmutableList - oak-core

* OAK-11311: Remove usage of Guava ImmutableList - oak-commons

* OAK-11311: Remove usage of Guava ImmutableList - oak-blob-plugins

* OAK-11311: Remove usage of Guava ImmutableList - oak-blob-cloud

* OAK-11311: Remove usage of Guava ImmutableList - oak-benchmarks

* OAK-11311: Remove usage of Guava ImmutableList - oak-authorization-principalbased

* OAK-11311: Remove usage of Guava ImmutableList - oak-auth-external

* OAK-11311: Remove usage of Guava ImmutableList - oak-doc

* OAK-11311: remove unused imports

Commit:
OAK-11335 : removed usage of Guava's Lists.asList with Stream API

kwin committed:
OAK-11319 Always execute apache-rat-plugin (#1912)

Remove "rat" profile
Enable profile "javadoc" for GHA CI

kwin committed:
OAK-11269 Commit Message Check (#1913)

Revert "Revert "OAK-11269 add dedicated workflow for checking commit messages (#1865)"
Add license headers to newly added files

reschke committed:
OAK-11269: commit message check - allow format commonly used in this project (#1938)

reschke committed:
OAK-11341: webapp: update Tomcat dependency to 9.0.98 (#1936)

reschke committed:
OAK-11342: update Tomcat JDBC dependency to 9.0.98 (#1940)

reschke committed:
OAK-11343: Update commons-io dependency to 2.18.0 (#1941)

Commit:
OAK-11333 : removed usage of Guava's Lists.transform with Stream API

reschke committed:
OAK-11300: shaded-guava: remove exports for packages that are not used (anymore) (#1898)

* OAK-11300: shaded-guava: remove exports for packages that are not used (anymore) - eventbus, html, xml

* OAK-11300: shaded-guava: remove exports for packages that are not used (anymore) - reflect

* OAK-11300: shaded-guava: remove exports for packages that are not used (anymore) - net

* OAK-11300: shaded-guava: remove exports for packages that are not used (anymore) - annotations

Commit:
OAK-11336 : removed usage of Guava's Lists

reschke committed:
OAK-11344: oak-benchmarks/oak-it-osgi: update commons-compress dependency to 1.27.1 (#1945)

mbaedke committed:
OAK-11199: Java 23: getSubject is supported only if a security manager is allowed

Removed redundant double dependency.

Commit:
OAK-11348 : removed usage of Guava's MapS.unmodifiableNavigableMap

reschke committed:
OAK-11314: Deprecate oak-solr for eventual removal (#1943)

* OAK-11314: Deprecate oak-solr for eventual removal

* Revert "OAK-11314: Deprecate oak-solr for eventual removal"

This reverts commit 3c3bb220ab85d4e6de708a2ee8a52c3568dc8944.

* OAK-11314: Deprecate oak-solr for eventual removal

Commit:
OAK-11349 : removed usage of Guava's Maps.newIdentityHashMap

Commit:
OAK-11353 : removed usage of Guava's Maps.newHashMap

reschke committed:
OAK-11001: Update commons-lang3 dependency to 3.17.0 (#1952)

Commit:
OAK-11347 : removed usage of Guava's Maps.fromProperties

reschke committed:
OAK-11354: Remove usage of Guava Ints.compare() (#1951)

reschke committed:
OAK-11359: Remove usage of Guava Ints.checkedCast (#1957)

nfsantos committed:
OAK-11356 - Code cleanups to indexing module (#1954)

rishabhdaim committed:
OAK-11350 : removed usage of Guava's Maps.filterKeys (#1956)

* OAK-11350 : removed usage of Guava's Maps.filterKeys

* OAK-11350 : fixed test failures due to insertion order mismatch

---------

Co-authored-by: Rishabh Kumar <diam@adobe.com>

mbaedke committed:
OAK-11284: Greedy Reuse of cluster IDs may lead to synchronous LastRe… (#1948)

Added documentation.

---------

Co-authored-by: Julian Reschke <reschke@apache.org>

reschke committed:
OAK-11361: Update commons-codec dependency to 1.17.2 (#1960)

fabriziofortino committed:
OAK-11352 (oak-search-elastic) add useInFullTextQuery flag for dynamic boosted values (#1949)

* OAK-11352 (oak-search-elastic) add useInFullTextQuery flag for dynamic boosted values

* OAK-11352 (minor) make useInFullTextQuery final

* OAK-11352 (test) fix dynamicBoostNotIncludedInFullText

reschke committed:
OAK-11360: remove usage of Guava Ints.contains() (#1958)

* OAK-11360: remove usage of Guava Ints.contains()

* OAK-11360: remove usage of Guava Ints.contains()

rishabhdaim committed:
OAK-11351 : removed usage of Guava's Maps.filterValues (#1959)

Co-authored-by: Rishabh Kumar <diam@adobe.com>

Commit:
OAK-11362 : removed usage of Guava's Maps.filterEntries

reschke committed:
OAK-11367: Remove usage of Guava Chars.contains() (#1964)

Commit:
OAK-11363 : removed usage of Guava's Maps.uniqueIndex

Commit:
OAK-11363 : replaced x -> x with Function.identity()

reschke committed:
OAK-11366: Remove usage of Guava UnsignedBytes.compare() (#1963)

* OAK-11366: Remove usage of Guava UnsignedBytes.compare()

* Update oak-segment-tar/src/main/java/org/apache/jackrabbit/oak/segment/SegmentVersion.java

Co-authored-by: Rishabh Kumar <rishabhdaim1991@gmail.com>

* OAK-11366: Remove usage of Guava UnsignedBytes.compare()

---------

Co-authored-by: Rishabh Kumar <rishabhdaim1991@gmail.com>

reschke committed:
OAK-11300: shaded-guava: remove exports for packages that are not used (anymore) - primitives

reschke committed:
Revert "OAK-11300: shaded-guava: remove exports for packages that are not used (anymore) - primitives"

This reverts commit dde8d90c3e33bcfa3dcb2f9ce61e1a0c7da340b0.

reschke committed:
OAK-11372: Release Oak 1.74.0 - Candidate Release Notes

Commit:
OAK-11374 : removed usage of Guava's Maps.immutableEntry

Commit:
OAK-11375 : removed usage of Guava's Maps.transformValues

Commit:
OAK-11375 : made inplace changes and removed the util method

reschke committed:
OAK-11377: Remove usage of Guava ByteStreams.toByteArray() (#1971)

* OAK-11377: Remove usage of Guava ByteStreams.toByteArray()

* OAK-11377: Remove usage of Guava ByteStreams.toByteArray() - avoid deprecated method

* Revert "OAK-11377: Remove usage of Guava ByteStreams.toByteArray() - avoid deprecated method"

This reverts commit d210344be29f8b855270e338fd08f64a60ce102f.

(as this change is beyond the scope of this PR)

jelmini committed:
OAK-11379 Ignore deleted ordered children in JsonSerializer output (#1974)

rishabhdaim committed:
OAK-11381 : removed usage of Guava's Maps.asMap (#1972)

Co-authored-by: Rishabh Kumar <diam@adobe.com>

rishabhdaim committed:
OAK-11382 : removed usage of Guava's Maps.toMap (#1973)

Co-authored-by: Rishabh Kumar <diam@adobe.com>

reschke committed:
OAK-11371: Remove usage of Guava UnsignedBytes.compare() (#1976)

reschke committed:
OAK-11372: Release Oak 1.74.0 - Candidate Release Notes

reschke committed:
[maven-release-plugin] prepare release jackrabbit-oak-1.74.0

reschke committed:
[maven-release-plugin] prepare for next development iteration

rishabhdaim committed:
OAK-11384 : removed usage of Guava's Maps.difference (#1977)

Co-authored-by: Rishabh Kumar <diam@adobe.com>

thomasmueller committed:
OAK-11365 Incremental index store: ability to set a timeout (#1962)

reschke committed:
OAK-11385: Remove usage of Guava Files.asByteSink (#1978)

reschke committed:
OAK-11357: Extract StreamUtils from CollectionUtils (#1980)

thomasmueller committed:
OAK-11027 flaky IncrementalStoreTest (#1979)

mreutegg committed:
OAK-11383: Improve performance of DefaultStatisticsProvider (#1975)

Get enumType only when necessary

reschke committed:
OAK-11380: Remove usage of Guava Files.asByteSource - oak-upgrade (#1982)

mbaedke committed:
OAK-11364: For oak-jcr tests, support automatic starting/stopping of … (#1961)

done

reschke committed:
OAK-11387: Extract ListUtils from CollectionUtils (#1983)

* OAK-11387: Extract ListUtils from CollectionUtils

* OAK-11387: Extract ListUtils from CollectionUtils - add class Javadoc

reschke committed:
OAK-11364: fix file mode

reschke committed:
OAK-11357: StreamUtils - add class javadoc

kwin committed:
Add badges to README

kwin committed:
OAK-11390 Generate aggregate UT/IT reports with profile "pedantic"

That helps to quickly check which tests take particularly long

rishabhdaim committed:
OAK-11395 : extracted out Set methods from CollectionUtils (#1988)

Co-authored-by: Rishabh Kumar <diam@adobe.com>

nfsantos committed:
OAK-11394 - Avoid creation of iterator when creating a CompositeEditor. (#1987)

rishabhdaim committed:
OAK-11388 : extracted out Map methods from CollectionUtils (#1989)

* OAK-11388 : extracted out Map methods from CollectionUtils

* OAK-11388 : removed closing of <p>

---------

Co-authored-by: Rishabh Kumar <diam@adobe.com>

rishabhdaim committed:
OAK-11396 : removed usage of Guava's Preconditions.checkElementIndex (#1990)

Co-authored-by: Rishabh Kumar <diam@adobe.com>

kwin committed:
OAK-11397 Map Oak to expanded JCR name in NameMapper

rishabhdaim committed:
OAK-11398 : removed usage of Guava's Preconditions.checkPositionIndexes (#1991)

Co-authored-by: Rishabh Kumar <diam@adobe.com>

kwin committed:
OAK-11397 Expose methods to get the expanded name/path from Item in
JackrabbitSession

rishabhdaim committed:
OAK-11400 : removed usage of Guava ArrayListMultimap (#1994)

Co-authored-by: Rishabh Kumar <diam@adobe.com>

kwin committed:
OAK-9500 Deploy Snapshot artifacts from trunk to ASF Nexus (#1935)

reschke committed:
OAK-11403: Remove usage of Guava io.ByteSource

reschke committed:
OAK-11403: Remove usage of Guava io.ByteSource

reschke committed:
OAK-11399: Remove usage of Guava io.BaseEncoding (#1992)

* OAK-11399: Remove usage of Guava io.BaseEncoding

* OAK-11399: Remove usage of Guava io.BaseEncoding - restore base32 case

* OAK-11399: Remove usage of Guava io.BaseEncoding

* OAK-11399: Remove usage of Guava io.BaseEncoding

rishabhdaim committed:
OAK-11402 : removed usage of Guava Strings.isNullOrEmpty (#1995)

* OAK-11402 : removed usage of Guava Strings.isNullOrEmpty

* OAK-11402 : added commons-lang3 in fix oak-store-composite ITs

---------

Co-authored-by: Rishabh Kumar <diam@adobe.com>

reschke committed:
OAK-11408: Extract IteratorUtils from CollectionUtils (#2003)

* OAK-11408: Extract IteratorUtils from CollectionUtils

* Update oak-commons/src/test/java/org/apache/jackrabbit/oak/commons/collections/IteratorUtilsTest.java

* OAK-11408: Extract IteratorUtils from CollectionUtils - address Sonar warnings

mbaedke committed:
OAK-11402: Remove usage of Guava Strings.isNullOrEmpty

Removed unused imports.

rishabhdaim committed:
OAK-11407 : removed usage of Guava Strings.EMPTY_STRING (#2004)

* OAK-11407 : removed usage of Guava Strings.EMPTY_STRING

* OAK-11407 : removed usage of Guava Strings.EMPTY_STRING with

---------

Co-authored-by: Rishabh Kumar <diam@adobe.com>

reschke committed:
OAK-11406: Remove usage of Guava io.ByteSource (#2006)

* OAK-11403: Remove usage of Guava io.ByteSource

* OAK-11406: Remove usage of Guava io.ByteSource

* OAK-11406: Remove usage of Guava io.ByteSource

* OAK-11406: Remove usage of Guava io.ByteSource

* OAK-11406: Remove usage of Guava io.ByteSource

* OAK-11406: Remove usage of Guava io.ByteSource - overrides

* OAK-11406: Remove usage of Guava io.ByteSource - remove unused code

reschke committed:
OAK-11415: TypeEditor: constraint exception for node type violations should contain that node's effective type (#2009)

rishabhdaim committed:
OAK-11413 : extracted out Deque methods from CollectionUtils (#2007)

* OAK-11413 : extracted out Deque methods from CollectionUtils

* OAK-11413 : added missing license header

---------

Co-authored-by: Rishabh Kumar <diam@adobe.com>

nfsantos committed:
OAK-11412 - Indexing job: delay creation of index writers (Lucene and Elastic) until the start of the indexing phase (#2008)

rishabhdaim committed:
OAK-11414 : making collectionutils package private (#2013)

Co-authored-by: Rishabh Kumar <diam@adobe.com>

rishabhdaim committed:
OAK-11411 : removed usage of Guava Strings.emptyToNull (#2015)

Co-authored-by: Rishabh Kumar <diam@adobe.com>

reschke committed:
OAK-11417: Remove usage of Guava ByteStreams  (#2014)

* OAK-11417: Remove usage of Guava ByteStreams

* OAK-11417: Remove usage of Guava ByteStreams - oak-blob-plugins

* OAK-11417: Remove usage of Guava ByteStreams - oak-lucene

* OAK-11417: Remove usage of Guava ByteStreams - oak-segment-tar

rishabhdaim committed:
OAK-11422 : removed usage of Guava Strings (#2016)

Co-authored-by: Rishabh Kumar <diam@adobe.com>

reschke committed:
OAK-11419: oak-store-document - persistent cache: remove unused UDP/TCP broadcasters (#2012)

* OAK-11419: oak-store-document - persistent cache: remove unused UDP broadcaster

* OAK-11419: oak-store-document - persistent cache: remove unused TCP broadcaster

reschke committed:
OAK-11425: MarkSweepGarbageColllector.iterateNodeTree() passes open file to FileIOUtils.sort() (#2017)

mbaedke committed:
OAK-11416: Docker support for oak-jcr tests with RdbDocumentStore nee… (#2010)

Implemented.

mbaedke committed:
OAK-11418: RdbConnectionUtils should use SystemPropertySupplier (#2011)

Done.

reschke committed:
OAK-11420: Remove usage of Guava Files.move() (#2019)

* OAK-11420: Remove usage of Guava Files.move()

* OAK-11420: Remove usage of Guava Files.move()

rishabhdaim committed:
OAK-11424 : removed usage of Guava ObjectArrays (#2018)

Co-authored-by: Rishabh Kumar <diam@adobe.com>

rishabhdaim committed:
OAK-11427 : removed usage of Guava Iterables.isEmpty (#2023)

* OAK-11427 : removed usage of Guava Iterables.isEmpty

* OAK-11427 : fixed import issue

* OAK-11427 : used commons-collections4 IterableUtils.isEmpty

---------

Co-authored-by: Rishabh Kumar <diam@adobe.com>

reschke committed:
OAK-11428: Remove usage of Guava Files.copy() (#2022)

* OAK-11428: Remove usage of Guava Files.copy() - oak-blob-plugins

* OAK-11428: Remove usage of Guava Files.copy() - oak-lucene

rishabhdaim committed:
OAK-11401 : increased audit log level for VGC (#2024)

Co-authored-by: Rishabh Kumar <diam@adobe.com>

kwin committed:
OAK-11389 Remove m-checkstyle-p (#1984)

rishabhdaim committed:
OAK-11430 : removed usage of Guava Iterables.all (#2026)

Co-authored-by: Rishabh Kumar <diam@adobe.com>

rishabhdaim committed:
OAK-11432 : removed usage of Guava Iterables.size (#2028)

Co-authored-by: Rishabh Kumar <diam@adobe.com>

reschke committed:
OAK-11420: Remove usage of Guava Files.move() - changes undone

reschke committed:
OAK-11420: Remove usage of Guava Files.move() - oak-blob-plugins (#2029)

reschke committed:
OAK-11434: oak-run: DataStoreCheckCommand tries to move open files (#2030)

reschke committed:
OAK-11431: Remove usage of Guava Files.createTempDir()  (#2027)

* OAK-11431: Remove usage of Guava Files.createTempDir()

* OAK-11431: Remove usage of Guava Files.createTempDir() - oak-lucene

* OAK-11431: Remove usage of Guava Files.createTempDir() - oak-upgrade

* OAK-11431: Remove usage of Guava Files.createTempDir() - oak-segment-aws

* OAK-11431: Remove usage of Guava Files.createTempDir() - oak-run-commons

* OAK-11431: Remove usage of Guava Files.createTempDir() - oak-run

mbaedke committed:
OAK-11421: Update MySql JDBC driver version to com.mysql:mysql-connector-j:9.0.1 in profile rdb-mysql (#2021)

Done.

mbaedke committed:
OAK-11423: Jdbc driver dependencies need to have scope compile in order to be included in oak-run. (#2020)

Fixed.

reschke committed:
OAK-11286: blob-cloud, segment: update netty version to 4.1.115 (#1879)

* OAK-11286: blob-cloud, segment: update netty version to 4.1.115

* OAK-11286: blob-cloud, segment: update netty version to 4.1.115

Exclude netty from the baseline check.

---------

Co-authored-by: Manfred Baedke <manfred.baedke@gmail.com>

reschke committed:
OAK-11434: oak-run: DataStoreCheckCommand tries to move open files

reschke committed:
OAK-11420: Remove usage of Guava Files.move() (#2033)

* OAK-11420: Remove usage of Guava Files.move() - oak-blob-plugins

* OAK-11420: Remove usage of Guava Files.move()

mbaedke committed:
OAK-11264: CacheWarming tests seems flaky (#2034)

Tests will now be skipped if the cleanup is not responsive.

Amoratinos committed:
OAK-11409: Allow multiple cache expiration properties (#2005)

* OAK-11409: Allow multiple cache expiration properties

* remove extra space in CacheConfiguration.java

* OAK-11409: Remove non-related formatting

* OAK-11409: Remove non-related formatting

* OAK-11409: Remove non-related formatting

* OAK-11409: Fixed security-spi package version

---------

Co-authored-by: anchela <angela@apache.org>

reschke committed:
OAK-11440: Remove usage of Guava Files.touch() (#2035)

rishabhdaim committed:
OAK-11439 : set remaining full gc modes (#2036)

Co-authored-by: Rishabh Kumar <diam@adobe.com>

rishabhdaim committed:
OAK-11433 : added OSGi config for setting max revision age for full gc (#2032)

* OAK-11433 : added OSGi config for setting max revision age for full gc

* OAK-11433 : removed un-used var

---------

Co-authored-by: Rishabh Kumar <diam@adobe.com>

reschke committed:
OAK-11445: Remove usage of Guava Files.toString() (#2039)

* OAK-11445: Remove usage of Guava Files.toString() - oak-core

* OAK-11445: Remove usage of Guava Files.toString() - oak-pojosr

* OAK-11445: Remove usage of Guava Files.toString() - oak-run

rishabhdaim committed:
OAK-11439 : fixed trunk build (#2040)

Co-authored-by: Rishabh Kumar <diam@adobe.com>

lweitzendorf committed:
OAK-11287: Cleanup May Delete Referenced Segments (#1880)

* OAK-11287: Cleanup May Delete Referenced Segments

* detect missing GC Journal entries

* remove AbstractGarbageCollectionStrategy

* remove some unrelated changes

---------

Co-authored-by: Lucas Weitzendorf <lweitzendorf@adobe.com>

reschke committed:
OAK-11446: Remove usage of Guava Files.equal()  (#2042)

* OAK-11446: Remove usage of Guava Files.equal()

* OAK-11446: Remove usage of Guava Files.equal()

reschke committed:
OAK-11439: set remaining full gc modes - add support for NONE without warning message

rishabhdaim committed:
OAK-11441 : removed usage of Guava Iterables.addAll (#2037)

* OAK-11441 : removed usage of Guava Iterables.addAll

* OAK-11441 : removed usage of Guava Iterables.addAll with iterable.foreach()

---------

Co-authored-by: Rishabh Kumar <diam@adobe.com>

reschke committed:
Revert "OAK-11397 Expose methods to get the expanded name/path from Item in"

This reverts commit 8ce6c4edd2e94a55b93481579d515daba260db6a.

rishabhdaim committed:
OAK-11442 : removed usage of Guava Iterables.contains (#2047)

Co-authored-by: Rishabh Kumar <diam@adobe.com>

reschke committed:
Revert "OAK-11397 Map Oak to expanded JCR name in NameMapper"

This reverts commit 2502f4c343412c48ef2ae5de5fa589dfb28c12cd.

rishabhdaim committed:
OAK-11449 : removed usage of Guava Iterables.concat (#2048)

Co-authored-by: Rishabh Kumar <diam@adobe.com>

reschke committed:
OAK-11448: Remove usage of Guava Files.readLines()  (#2046)

* OAK-11448: Remove usage of Guava Files.readLines()

* OAK-11448: Remove usage of Guava Files.readLines()

* OAK-11448: Remove usage of Guava Files.readLines()

* OAK-11448: Remove usage of Guava Files.readLines()

* OAK-11448: Remove usage of Guava Files.readLines()

kwin committed:
OAK-11450 Correctly detect expanded names using empty namespace or
internal namespace

rishabhdaim committed:
OAK-11453 : added support for setting max revision age for full gc fr… (#2051)

* OAK-11453 : added support for setting max revision age for full gc from oak-run

* OAK-11453 : changed variable name for setting full gc max revision age

---------

Co-authored-by: Rishabh Kumar <diam@adobe.com>

reschke committed:
OAK-11455: Release Oak 1.76.0 - Candidate Release Notes

reschke committed:
[maven-release-plugin] prepare release jackrabbit-oak-1.76.0

reschke committed:
[maven-release-plugin] prepare for next development iteration

thomasmueller committed:
OAK-11454 Explain query output: fix synchronousPropertyCondition (#2052)

* OAK-11454 Explain query output: fix synchronousPropertyCondition

* OAK-11454 Explain query output: fix synchronousPropertyCondition

thomasmueller committed:
OAK-11454 Explain query output: fix synchronousPropertyCondition (#2052)

* OAK-11454 Explain query output: fix synchronousPropertyCondition

* OAK-11454 Explain query output: fix synchronousPropertyCondition

thomasmueller committed:
OAK-11457 Tree store sometimes contains bundled properties (#2053)

* OAK-11457 Tree store sometimes contains bundled properties

* OAK-11457 Tree store sometimes contains bundled properties

thomasmueller committed:
OAK-11457 Tree store sometimes contains bundled properties (#2053)

* OAK-11457 Tree store sometimes contains bundled properties

* OAK-11457 Tree store sometimes contains bundled properties

nfsantos committed:
OAK-11459 - Code cleanups on indexing related packages (#2056)

nfsantos committed:
OAK-11459 - Code cleanups on indexing related packages (#2056)

ierandra committed:
OAK-8413 Use the new Azure SDK in the Azure Segment Store (#1748)

* OAK-8413 Use the new Azure SDK in the Azure Segment Store

* OAK-8413 Use the new Azure SDK in the Azure Segment Store - configure package versions

* OAK-8413 Use the new Azure SDK in the Azure Segment Store - PR review fixes

* OAK-8413 Use the new Azure SDK in the Azure Segment Store - fix tests

* OAK-8413 Use the new Azure SDK in the Azure Segment Store - fix tests

* OAK-8413 Use the new Azure SDK in the Azure Segment Store - remove debug logging

* OAK-8413 Use the new Azure SDK in the Azure Segment Store - PR review fixes

* OAK-8413 Use the new Azure SDK in the Azure Segment Store - fix rootPrefix and missing metadata

* OAK-8413 Use the new Azure SDK in the Azure Segment Store - fix rootPrefix issue

* OAK-8413 Use the new Azure SDK in the Azure Segment Store - PR review

* OAK-8413 Use the new Azure SDK in the Azure Segment Store - fix rootPrefix issue

* OAK-8413 Use the new Azure SDK in the Azure Segment Store - PR review

* OAK-8413 Use the new Azure SDK in the Azure Segment Store - PR review

* OAK-8413 Use the new Azure SDK in the Azure Segment Store - PR review

* OAK-8413 Use the new Azure SDK in the Azure Segment Store - PR merge

* OAK-8413 Use the new Azure SDK in the Azure Segment Store - PR merge

* OAK-8413 Use the new Azure SDK in the Azure Segment Store - PR review

* OAK-8413 Use the new Azure SDK in the Azure Segment Store - fix tests

* OAK-8413 Use the new Azure SDK in the Azure Segment Store - fix tests

* Updates from master

ierandra committed:
OAK-8413 Use the new Azure SDK in the Azure Segment Store (#1748)

* OAK-8413 Use the new Azure SDK in the Azure Segment Store

* OAK-8413 Use the new Azure SDK in the Azure Segment Store - configure package versions

* OAK-8413 Use the new Azure SDK in the Azure Segment Store - PR review fixes

* OAK-8413 Use the new Azure SDK in the Azure Segment Store - fix tests

* OAK-8413 Use the new Azure SDK in the Azure Segment Store - fix tests

* OAK-8413 Use the new Azure SDK in the Azure Segment Store - remove debug logging

* OAK-8413 Use the new Azure SDK in the Azure Segment Store - PR review fixes

* OAK-8413 Use the new Azure SDK in the Azure Segment Store - fix rootPrefix and missing metadata

* OAK-8413 Use the new Azure SDK in the Azure Segment Store - fix rootPrefix issue

* OAK-8413 Use the new Azure SDK in the Azure Segment Store - PR review

* OAK-8413 Use the new Azure SDK in the Azure Segment Store - fix rootPrefix issue

* OAK-8413 Use the new Azure SDK in the Azure Segment Store - PR review

* OAK-8413 Use the new Azure SDK in the Azure Segment Store - PR review

* OAK-8413 Use the new Azure SDK in the Azure Segment Store - PR review

* OAK-8413 Use the new Azure SDK in the Azure Segment Store - PR merge

* OAK-8413 Use the new Azure SDK in the Azure Segment Store - PR merge

* OAK-8413 Use the new Azure SDK in the Azure Segment Store - PR review

* OAK-8413 Use the new Azure SDK in the Azure Segment Store - fix tests

* OAK-8413 Use the new Azure SDK in the Azure Segment Store - fix tests

* Updates from master

mbaedke committed:
OAK-11435: oak-search: the special baseline configuration may be removed from the pom. (#2044)

done

mbaedke committed:
OAK-11435: oak-search: the special baseline configuration may be removed from the pom. (#2044)

done

mbaedke committed:
OAK-11436: oak-segment-aws: the special baseline configuration may be removed from the pom. (#2043)

done

mbaedke committed:
OAK-11436: oak-segment-aws: the special baseline configuration may be removed from the pom. (#2043)

done

mbaedke committed:
OAK-11456: oak-solr-osgi embeds vulnerable Zookeeper 3.9.2 (#2061)

Updated to 3.6.3.

mbaedke committed:
OAK-11456: oak-solr-osgi embeds vulnerable Zookeeper 3.9.2 (#2061)

Updated to 3.6.3.

reschke committed:
Revert "OAK-8413 Use the new Azure SDK in the Azure Segment Store (#1748)"

This reverts commit a8fe2ac814a0064af68bd81801040eb1edf70a09.

reschke committed:
Revert "OAK-11459 - Code cleanups on indexing related packages (#2056)"

This reverts commit 080a08b806bafd1bcec1c6c6659be990c29116e2.

reschke committed:
Revert "OAK-11456: oak-solr-osgi embeds vulnerable Zookeeper 3.9.2 (#2061)"

This reverts commit 1d9507d4aa10c9a5f21bea8997ca87055a352ac9.

reschke committed:
Revert "OAK-11436: oak-segment-aws: the special baseline configuration may be removed from the pom. (#2043)"

This reverts commit 4f84b5cee375a2400569e5cbf47b6d41878c6ba6.

reschke committed:
Revert "OAK-11457 Tree store sometimes contains bundled properties (#2053)"

This reverts commit 7d46f395f55bcf3b5c2815ed24e3ef754a6e3b84.

reschke committed:
Revert "OAK-11454 Explain query output: fix synchronousPropertyCondition (#2052)"

This reverts commit 7d241126ee60c937b814c61fb9ccf869192478d3.

reschke committed:
Revert "OAK-11435: oak-search: the special baseline configuration may be removed from the pom. (#2044)"

This reverts commit 9365bd5683a8e78d2fc011f07efe282686aa5f64.

reschke committed:
Revert "[maven-release-plugin] prepare for next development iteration"

This reverts commit 5f5851e69e85563c7d73505cbaafbf84d495f9ba.

reschke committed:
Revert "[maven-release-plugin] prepare release jackrabbit-oak-1.76.0"

This reverts commit dc9c36c794022bbeccea3b344937ce29cff86f1c.

reschke committed:
OAK-11470: Undo changes to remove usage of Guava ImmutableSet (#2066)

* OAK-11465: undo non-test changes done in OAK-11317

* OAK-11465: undo non-test changes done in OAK-11317

* OAK-11465: undo non-test changes done in OAK-11317

* OAK-11465: undo non-test changes done in OAK-11317

* set fake version string

reschke committed:
OAK-11470: Undo changes to remove usage of Guava ImmutableSet - restore changes in test classes

reschke committed:
OAK-11455: Release Oak 1.76.0 - Candidate Release Notes

reschke committed:
OAK-11455: Release Oak 1.76.0 - Candidate Release Notes

reschke committed:
[maven-release-plugin] prepare release jackrabbit-oak-1.76.0

reschke committed:
[maven-release-plugin] prepare for next development iteration

nfsantos committed:
OAK-11469 - Speedup resolution of mount paths by SimpleMountInfoProvider (#2065)

thomasmueller committed:
OAK-11460 Allow configuring the index.mapping.total_fields.limit (#2058)

* OAK-11460 Allow configuring the index.mapping.total_fields.limit

* OAK-11460 Allow configuring the index.mapping.total_fields.limit

niekraaijmakers committed:
OAK-11461 : DocumentNodeStoreService -  disable discovery  (#2057)

* OAK-11461 - Make it configurable to disable discovery for the document node store service

* Update oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/Configuration.java

Co-authored-by: Rishabh Kumar <rishabhdaim1991@gmail.com>

* OAK-11461 - Add unit test

* Update oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/Configuration.java

Co-authored-by: José Andrés Cordero Benítez <Joscorbe@users.noreply.github.com>

---------

Co-authored-by: Niek Raaijmakers <raaijmak@adobe.com>
Co-authored-by: Rishabh Kumar <rishabhdaim1991@gmail.com>
Co-authored-by: José Andrés Cordero Benítez <Joscorbe@users.noreply.github.com>

nfsantos committed:
OAK-11472 - Better error message when Tika configuration is missing. (#2068)

reschke committed:
OAK-11455: Release Oak 1.76.0 - POM cleanup

reschke committed:
OAK-11462: Remove usage of Guava Files.write() (#2059)

* OAK-11462: Remove usage of Guava Files.write() -  oak-core

* OAK-11462: Remove usage of Guava Files.write() -  oak-run

* OAK-11462: Remove usage of Guava Files.write() -  oak-upgrade

* OAK-11462: Remove usage of Guava Files.write() - whitespace fixed

reschke committed:
OAK-11481: Remove (once more) usage of Guava Files.createTempDir() (#2074)

reschke committed:
OAK-11482: Remove usage of Guava Files.newReader (#2075)

fabriziofortino committed:
OAK-11475: fix param conversion for WordDelimiterFilter (#2070)

* tests: filters should always be ordered

* OAK-11475: fix param conversion for WordDelimiterFilter

thomasmueller committed:
OAK-11488 Flaky test org.apache.jackrabbit.oak.plugins.document.VersionGarbageCollectorIT. testDeletedPropsAndUnmergedBCWithCollisionWithDryRunMode (#2083)

thomasmueller committed:
OAK-11489 Flaky test BranchCommitGCTest.unmergedAddsThenMergedAddThenUnmergedRemovesChildren (#2084)

mbaedke committed:
OAK-11464: oak-it-osgi fails to load oak-search-elastic if the profile rdb-mysql is used for building and testing. (#2060)

Removed package import.

reschke committed:
OAK-11484: Remove usage of Guava io.CountingInputStream (#2077)

thomasmueller committed:
OAK-11490 VersionGarbageCollectorIT takes 1 hour to run - ignore for now (#2087)

smiroslav committed:
OAK-11491 clear system property in tearDown method (#2086)

* OAK-11491 clear system property in tearDown method

* OAK-11491 remember the old value of the system property

* OAK-11491 rename property

---------

Co-authored-by: Miroslav Smiljanic <miroslav@apache.com>

ahanikel committed:
OAK-11480 - Graceful handling when getErrorCode() returns null. (#2078)

Co-authored-by: Axel Hanikel <ahanikel@adobe.com>

nfsantos committed:
OAK-11483 - Throttle log warn messages during indexing in Elastic (#2080)

nfsantos committed:
OAK-11438 - Add support for parallelization in the Lucene index writer backend. (#2031)

rishabhdaim committed:
OAK-11458 : added utils class for replacing Guava's Iterables (#2055)

* OAK-11458 : added utils class for repalcing Guava's Iterables

* OAK-11458 : fixed sonarcloud alerts by calling apache commons-collections method directly from each API

---------

Co-authored-by: Rishabh Kumar <diam@adobe.com>

stefan-egli committed:
OAK-11477 : dont fail if lease couldnt be renewed in crash simulation (#2088)

Fixed test case.

rishabhdaim committed:
OAK-11494 : added utils class for repalcing Guava's Iterables.contains (#2089)

Co-authored-by: Rishabh Kumar <diam@adobe.com>

rishabhdaim committed:
OAK-11495 : added utils method for repalcing Guava's Iterables.size (#2090)

Co-authored-by: Rishabh Kumar <diam@adobe.com>

rishabhdaim committed:
OAK-11496 : added utils method for repalcing Guava's Iterables.all (#2096)

Co-authored-by: Rishabh Kumar <diam@adobe.com>

mbaedke committed:
OAK-11468: RDBDataSourceFactory unnecessarily loads the JDBC driver class explicitly. (#2085)

Removed unnecessary code.

rishabhdaim committed:
OAK-11503 : added utils method for repalcing Guava's Iterables.size (#2097)

Co-authored-by: Rishabh Kumar <diam@adobe.com>

amit-jain commented on oak-blob-cloud-azure/src/main/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/AzureBlobContainerProvider.java:
```diff
@@ -171,176 +153,92 @@ public String getContainerName() {
-            }
-        }
+    private ClientSecretCredential getClientSecretCredential() {
+        return new ClientSecretCredentialBuilder()
```
Does this not have any expiry issues? 
As I understand earlier the problem was that the access token retrieved expired in 60 minutes and that is why TokenRefresher functionality was added.

ierandra replied:
With Azure skd v12 this is done automatically so this is not an issue anymore.

amit-jain commented on oak-blob-cloud-azure/src/main/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/AzureBlobStoreBackend.java:
```diff
@@ -1264,34 +1133,22 @@ private boolean loadItems() {
-                CloudBlobContainer container = azureBlobContainerProvider.getBlobContainer();
-                if (!firstCall && (resultContinuation == null || !resultContinuation.hasContinuation())) {
-                    LOG.trace("No more records in container. containerName={}", container);
+                if (!firstCall) {
```
The pagination has been removed. In this case filling up a list in one go will potentially cause OOM. 
The result object is a PagedIterable and getting an iterator() ans delegating to it should be much more simpler.

amit-jain replied:
Ref https://learn.microsoft.com/en-us/java/api/com.azure.storage.blob.models?view=azure-java-stable

amit-jain commented on oak-blob-cloud-azure/src/main/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/AzureBlobStoreBackend.java:
```diff
@@ -1362,4 +1219,54 @@ private String getContainerName() {
+        return key;
+    }
+
+    private String computeSecondaryLocationEndpoint() {
```
Any documentation of why this is needed will be helpful

amit-jain commented on oak-blob-cloud-azure/src/main/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/Constants.java:
```diff
@@ -0,0 +1,19 @@
+    long BUFFERED_STREAM_THRESHOLD = 1024 * 1024;
+    long MIN_MULTIPART_UPLOAD_PART_SIZE = 1024 * 1024 * 10; // 10MB
+    long MAX_MULTIPART_UPLOAD_PART_SIZE = 1024 * 1024 * 100; // 100MB
+    long MAX_SINGLE_PUT_UPLOAD_SIZE = 1024 * 1024 * 256; // 256MB, Azure limit
```
Not super important but are these the latest limits which are documented here?

amit-jain commented on oak-blob-cloud-azure/src/test/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/AzureBlobStoreBackendTest.java:
```diff
@@ -18,8 +18,10 @@
  */
 package org.apache.jackrabbit.oak.blob.cloud.azure.blobstorage;
 
-import com.microsoft.azure.storage.StorageException;
```
These tests should be moved to the v8 package as we still have the code living in the v8 package

amit-jain commented on oak-jcr/src/test/java/org/apache/jackrabbit/oak/jcr/binary/fixtures/datastore/AzureDataStoreFixture.java:
```diff
@@ -94,7 +95,7 @@ public DataStore createDataStore() {
         String connectionString = Utils.getConnectionStringFromProperties(azProps);
         try {
-            CloudBlobContainer container = Utils.getBlobContainer(connectionString, containerName);
+            CloudBlobContainer container = UtilsV8.getBlobContainer(connectionString, containerName);
```
Should we have a property to enable/disable v8/v12 so that both can be used.

amit-jain commented on oak-run-commons/src/main/java/org/apache/jackrabbit/oak/fixture/DataStoreUtils.java:
```diff
@@ -188,7 +189,7 @@ private static CloudBlobContainer getCloudBlobContainer(@NotNull Map<String, ?>
         }
 
-        try (AzureBlobContainerProvider azureBlobContainerProvider = AzureBlobContainerProvider.Builder.builder(containerName)
+        try (AzureBlobContainerProviderV8 azureBlobContainerProvider = AzureBlobContainerProviderV8.Builder.builder(containerName)
```
Would be better to have an option and support both

amit-jain commented on oak-run-commons/src/test/java/org/apache/jackrabbit/oak/fixture/DataStoreUtilsTest.java:
```diff
@@ -151,7 +152,7 @@ public void delete_container_service_principal() throws Exception {
 
         CloudBlobContainer container;
-        try (AzureBlobContainerProvider azureBlobContainerProvider = AzureBlobContainerProvider.Builder.builder(CONTAINER_NAME)
+        try (AzureBlobContainerProviderV8 azureBlobContainerProvider = AzureBlobContainerProviderV8.Builder.builder(CONTAINER_NAME)
```
same as above, support for v12

amit-jain reviewed (changes requested):
Grnerally looks good to me. But need resolution on pagination on listing.

rishabhdaim committed:
OAK-11451 : removed usage of Guava Iterables.concat (#2049)

* OAK-11451 : removed usage of Guava Iterables.concat

* OAK-11451 : fixed test failure caused due to collecting element into list via Stream

* OAK-11451 : fixed test failure because streams can't be used twice

* OAK-11451 : reverted code changes for Iterables.concat with Iterable<Iterable<E>>

* OAK-11451 : reverted code changes for Iterables.concat with Iterable<Iterable<E>> from oak-segment

* OAK-11451 : reverted unused imports

* OAK-11451 : removed usage of Guava Iterables.concat with oak-commons IterableUtils

---------

Co-authored-by: Rishabh Kumar <diam@adobe.com>

rishabhdaim committed:
OAK-11507 : using oak-commons IterableUtils inplace of commons-collections4 (#2100)

* OAK-11507 : using oak-commons IterableUtils inplace of commons-collections4

* OAK-11507 : did reformatting

---------

Co-authored-by: Rishabh Kumar <diam@adobe.com>

fabriziofortino committed:
OAK-11508: [oak-search-elastic] is null / is not null queries should resolve the field name (#2101)

* OAK-11508: [oak-search-elastic] is null / is not null queries should resolve the field name

* OAK-11508: [oak-search-elastic] is null / is not null queries should resolve the field name

thomasmueller committed:
OAK-11252 Disabling flaky test (#2103)

thomasmueller committed:
OAK-11509 Incremental flat file store: reduce logging (#2102)

* OAK-11509 Incremental flat file store: reduce logging

* OAK-11509 Incremental flat file store: reduce logging

* OAK-11509 Incremental flat file store: reduce logging

thomasmueller committed:
OAK-11504 Elasticsearch: support flattened fields - indentation changes only (#2104)

nfsantos committed:
OAK-11515 - Fix indentation and import order (#2106)

rishabhdaim committed:
OAK-11513 : added util method in oak-commons to replace Iterables.toArray (#2105)

Co-authored-by: Rishabh Kumar <diam@adobe.com>

rishabhdaim committed:
OAK-11513 : removed usage of Guava's Iterables.toArray (#2107)

Co-authored-by: Rishabh Kumar <diam@adobe.com>

ierandra committed:
OAK-11466 Replace Fixed to Exponential retries in oak-segment-azure (#2067)

* OAK-11466 Replace Fixed to Exponential retries in oak-segment-azure

* OAK-11466 Replace Fixed to Exponential retries in oak-segment-azure

* OAK-11466 Replace Fixed to Exponential retries in oak-segment-azure

reschke committed:
OAK-11397: add new API for expanded names/paths  (#2079)

Co-authored-by: Konrad Windszus <kwin@apache.org>

nfsantos committed:
OAK-11505 - Reduce object allocation in FullTextIndexEditor (#2098)

rishabhdaim committed:
OAK-11518 : added utils method to replace Iterables.partition (#2111)

* OAK-11518 : added utils method to replace Iterables.partition

* OAK-11518 : updated return type to iterable

* OAK-11518 : added requireNotNull and checkargument check

* OAK-11518 : throws an exception if no element is present while calling Iterable.next()

* Update IterableUtilsTest.java

---------

Co-authored-by: Rishabh Kumar <diam@adobe.com>
Co-authored-by: mbaedke <manfred.baedke@gmail.com>

rishabhdaim committed:
OAK-11519 : removed usage of Iterables.partition with oak-commons (IterableUtils) (#2112)

Co-authored-by: Rishabh Kumar <diam@adobe.com>

rishabhdaim committed:
OAK-11523 : added util method for replceing Guava's Iterables.filter … (#2114)

* OAK-11523 : added util method for replceing Guava's Iterables.filter in oak-commons module

* OAK-11523 : added util method for replacing Iterables.filter's Class Type API

---------

Co-authored-by: Rishabh Kumar <diam@adobe.com>

reschke commented:
This PR needs to be rebased. It currently has >1000K files changed.

andreeastroe96 commented:
I will close a PR as I am not longer the owner of this task
